### PR TITLE
use table_set_width(table, 0) if --full is specified

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/analyze/analyze-security.c [hunk 1]
```diff
@@ -1507,21 +1506,20 @@ static int assess(const struct security_info *info, Table *overview_table, Analy
                         if (d)
                                 description = d;
 
-                        r = table_add_cell_full(details_table, &cell, TABLE_STRING, checkmark, 1, 1, 0, 0, 0);
+                        r = table_add_cell_full(details_table, NULL, TABLE_STRING, checkmark, 1, 1, 0, 0, 0);
```
Comment (keszybz) [inline src/analyze/analyze-security.c]: I get the following crash from `build/systemd-analyze security`:
```
Assertion 'first_type < _TABLE_DATA_TYPE_MAX' failed at src/shared/format-table.c:741, function table_add_many_internal(). Abortin>

Program received signal SIGABRT, Aborted.
0x00007ffff79f0625 in raise () from /lib64/libc.so.6
Missing separate debuginfos, use: dnf debuginfo-install audit-libs-3.0-0.15.20191104git1c2f876.fc31.x86_64 cryptsetup-libs-2.2.2-1.fc31.x86_64 device-mapper-libs-1.02.165-1.fc31.x86_64 iptables-libs-1.8.3-5.fc31.x86_64 json-c-0.13.1-8.fc31.x86_64 kmod-libs-26-4.fc31.x86_64 libacl-2.2.53-4.fc31.x86_64 libargon2-20171227-3.fc31.x86_64 libattr-2.4.48-7.fc31.x86_64 libblkid-2.34-3.fc31.x86_64 libcap-2.26-6.fc31.x86_64 libcap-ng-0.7.10-1.fc31.x86_64 libffi-3.1-23.fc31.x86_64 libgcc-9.2.1-1.fc31.x86_64 libgcrypt-1.8.5-1.fc31.x86_64 libgpg-error-1.36-2.fc31.x86_64 libidn2-2.3.0-1.fc31.x86_64 libmount-2.34-3.fc31.x86_64 libpcap-1.9.1-1.fc31.x86_64 libseccomp-2.4.2-1.fc32.x86_64 libselinux-2.9-5.fc31.x86_64 libuuid-2.34-3.fc31.x86_64 lz4-libs-1.9.1-1.fc31.x86_64 openssl-libs-1.1.1d-2.fc31.x86_64 pam-1.3.1-18.fc31.x86_64 pcre2-10.33-16.fc31.x86_64 xz-libs-5.2.4-6.fc31.x86_64 zlib-1.2.11-20.fc31.x86_64
(gdb) bt
#0  0x00007ffff79f0625 in raise () from /lib64/libc.so.6
#1  0x00007ffff79d98d9 in abort () from /lib64/libc.so.6
#2  0x00007ffff7e195d8 in log_assert_failed_realm (realm=LOG_REALM_SYSTEMD, 
    text=0x7ffff7f2fa30 "first_type < _TABLE_DATA_TYPE_MAX", file=0x7ffff7f2f82a "src/shared/format-table.c", line=741, 
    func=0x7ffff7f30140 <__PRETTY_FUNCTION__.9038> "table_add_many_internal") at ../src/basic/log.c:809
#3  0x00007ffff7d77037 in table_add_many_internal (t=0x5fa180, first_type=TABLE_SET_COLOR) at ../src/shared/format-table.c:741
#4  0x000000000042a621 in assess (info=0x7fffffffd430, overview_table=0x0, flags=(unknown: 0))
    at ../src/analyze/analyze-security.c:1513
#5  0x000000000042c417 in analyze_security_one (bus=0x5f7d90, name=0x5f6f50 "wpa_supplicant.service", overview_table=0x0, 
    flags=(unknown: 0)) at ../src/analyze/analyze-security.c:2012
#6  0x000000000042ccfb in analyze_security (bus=0x5f7d90, units=0x7fffffffd908, flags=(unknown: 0))
    at ../src/analyze/analyze-security.c:2118
#7  0x00000000004258f9 in do_security (argc=2, argv=0x7fffffffd900, userdata=0x0) at ../src/analyze/analyze.c:2178
#8  0x00007ffff7dd6dbd in dispatch_verb (argc=3, argv=0x7fffffffd900, verbs=0x5469c0 <verbs>, userdata=0x0)
    at ../src/shared/verbs.c:112
#9  0x00000000004262e9 in run (argc=3, argv=0x7fffffffd8f8) at ../src/analyze/analyze.c:2468
#10 0x000000000042631c in main (argc=3, argv=0x7fffffffd8f8) at ../src/analyze/analyze.c:2471
```

File: src/machine/machinectl.c [hunk 2]
```diff
@@ -386,6 +389,9 @@ static int list_images(int argc, char *argv[], void *userdata) {
         if (!table)
                 return log_oom();
 
+        if (arg_full)
+                table_set_width(table, 0);
```
Comment (keszybz) [inline src/machine/machinectl.c:393]: I get the following:
```console
$ build/machinectl --full
MACHINE CLASS     SERVICE        OS     VERSION ADDRESSES       
rawhide container systemd-nspawn fedora 32      192.168.122.249…

1 machines listed.
```
... i.e. there's still ellipsization.

Comment (poettering) [reply src/machine/machinectl.c:393]: that ellipsation is not done by the table logic, but by machinectl itself, it only shows one ip address by default, see the --max-addresses=

One could argue though that --full should influence --max-addresses= too, but that#s probably a separate commit

Comment (poettering) [reply src/machine/machinectl.c:393]: also, the table formatting code should probably wisen up and support multi-line fields sanely, to cover for the IP address case better...

**Comment by yuwata** [comment]
@keszybz and @poettering Thank you for the comments. I've force-pushed the revised version. In the revised one, crash in systemd-analyze is fixed. PTAL.

**Comment by yuwata** [comment]
Also, the address truncation is addressed.

**Pull request merged**